clap_complete/src/shells/zsh.rs

```diff
@@ -622,18 +622,49 @@ fn write_positionals_of(p: &Command) -> String {
 
     let mut ret = vec![];
 
-    for arg in p.get_positionals() {
+    // The index of the 2nd-to-last positional argument.
+    let positionals: Vec<_> = p.get_positionals().collect();
+    let penultimate_idx = if positionals.len() > 1 {
+        Some(positionals.len() - 2)
+    } else {
+        None
+    };
+    // Completions for commands that end with two Vec arguments require special care:
+    // - You can have two Vec args separated with a value_terminator.
+    // - You can have two Vec args with the second one set to last (raw sets last)
+    //   which will force a separator of -- before the second.
+    // zsh requires a terminator in both of these situations.
+    let needs_terminator = match positionals.last() {
```

Comment:
If this is referring to `--`, I'd prefer it as `needs_escaping` as that is basically what `--` does and distinguishes it from `value_terminator`.

Reply:
That's a good point about the unclear terminology. The comment about zsh would probably read more clearly as:

```
// zsh requires a separator in both of these situations
```

basically, referring to it as a "separator" instead of a "terminator" to make it clear that they're two different concepts.

Ditto -- `needs_terminator` probably should be named `needs_escaping` or `needs_separator`.